src/domains/ecommerce-app/api/ecommerce-checkout/v1/_payment_request_policy.xml.tpl

```diff
@@ -21,9 +22,24 @@
     <set-variable name="recaptcha-check-body" value="@(((IResponse)context.Variables["recaptcha-check"]).Body.As<JObject>())" />
     <choose>
       <when condition="@(((IResponse)context.Variables["recaptcha-check"]).StatusCode != 200 || ((bool) ((JObject) context.Variables["recaptcha-check-body"])["success"]) != true)">
-        <return-response>
-          <set-status code="401" reason="Unauthorized" />
-          </return-response>
+          <!-- Double Check google reCAPTCHA Enterprise token validity START -->
+           <send-request ignore-error="true" timeout="10" response-variable-name="recaptcha-enterprise-check" mode="new">
```

Comment:
I would rather switch this "enterprise verification api call check" with the above one since once switch is performed is expected that all verifications will succeed with the enterprise (new) version and api call to the old one will be performed only for those api call coming from an un-updated checkout version (cached version maybe).
This way we will avoid a double api call for all the updated frontends.
What do you think?

Reply:
I thought about it, but this would mean applying the policy very close to the actual release. Otherwise, we would slightly affect the metrics on the new reCAPTCHA. So, we pay for an extra API call, but we don't impact the metrics on the new reCAPTCHA.

Reply:
I might not have been very clear :-) The idea is to apply it in production a few hours before switching the frontend... this way, we can be more confident when doing the switch with the frontend. I wouldn't want traffic to hit the new reCAPTCHA with tokens from the old reCAPTCHA, to avoid affecting the metrics of the new reCAPTCHA.